src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -683,15 +690,7 @@ void OnSubscriptionEstablished(uint64_t aSubscriptionId) override
         }
     }
 
-    void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override
-    {
-        VerifyOrDie(
-            aReadPrepareParams.mAttributePathParamsListSize == 1 && aReadPrepareParams.mpAttributePathParamsList != nullptr);
-        chip::Platform::Delete<app::AttributePathParams>(aReadPrepareParams.mpAttributePathParamsList);
-
-        VerifyOrDie(aReadPrepareParams.mDataVersionFilterListSize == 1 && aReadPrepareParams.mpDataVersionFilterList != nullptr);
-        chip::Platform::Delete<app::DataVersionFilter>(aReadPrepareParams.mpDataVersionFilterList);
-    }
+    void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override {}
```

Comment:
Attribute path params list is deleted from OnDone and Data version filter list is not set and hence both verifications were removed. Thanks @bzbarsky-apple for catching this.